crates/cairo-profiler/src/profile_builder.rs

```diff
@@ -66,32 +66,74 @@ impl ProfilerContext {
         }
     }
 
-    fn location_id(&mut self, location: &Function) -> LocationId {
-        let location = match location {
-            Function::Entrypoint(function_name) | Function::NonInlined(function_name) => {
-                function_name
+    fn locations_ids(&mut self, call_stack: &[Function]) -> Vec<LocationId> {
+        let mut locations_ids = vec![];
```

Comment:
No logic should be performed in `profile_builder` you should do this in previous layer and mark elements in a sample location as inlined.

Reply:
Done